src/features/filter.ts

```diff
@@ -1,10 +1,67 @@
-export const actions = { /* put action creators here */};
+type SetFilterAll = {
+  type: 'currentTodo/All',
+};
+
+type SetFilterActive = {
+  type: 'currentTodo/ACTIVE',
+};
+
+type SetFilterCompleted = {
+  type: 'currentTodo/COMPLETED',
+};
+
+type SetFilterQuery = {
+  type: 'currentTodo/SET_QUERY',
+  payload: string,
+};
+
+const setFilterQuery = (query: string): SetFilterQuery => ({
+  type: 'currentTodo/SET_QUERY',
+  payload: query,
+});
+
+const setActiveTodo = (): SetFilterActive => ({
+  type: 'currentTodo/ACTIVE',
+});
+
+const setCompletedTodo = (): SetFilterCompleted => ({
+  type: 'currentTodo/COMPLETED',
+});
+
+const setAllTodos = (): SetFilterAll => ({
+  type: 'currentTodo/All',
+});
```

Comment:
instead three action creators  you can make one action creator and pass payload (proper filter) from  component.
something like that
```
const setStatus = (filter: Filter): SetFilterAction => ({
  type: 'filter/SET_FILTER',
  payload: filter,
});
```

Reply:
and make enum for Filter and use it everywhere instead hardcoded strings
```
enum Filter {
  ALL = 'all',
  ACTIVE = 'active',
  COMPLETED = 'completed',
}